src/mono/mono/metadata/native-library.c

```diff
@@ -1262,6 +1264,13 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 	addr = pinvoke_probe_for_symbol (module, piinfo, new_import, &error_msg);
 
 	if (!addr) {
+#if defined(ENABLE_NETCORE) && !defined(HOST_WIN32)
+		if (strcmp (new_scope, "__Internal") == 0) {
```

Comment:
Is there a reason this is located here instead of with the other `__Internal` special-casing in `netcore_lookup_native_library`? I was thinking just create a static `MonoDl` the same way we do for `dlopen (NULL)`. This feels unnecessarily roundabout and separates related code.